### ISSUE TITLE
ViewColumn in Entity

### ISSUE THREAD

**Issue by Ben-Ho** [typeorm/typeorm#5139]
**Issue type:**

[ ] question
[ ] bug report
[x ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Description**
Use ViewColumn in Entity to add not persisted, dynamically calculated columns.

***Note***
I'm used to work with [koala-framework](https://github.com/koala-framework/koala-framework). It does a lot, but there is one handy feature called [stored model-expressions](http://docs.koala-framework.org/kwf-general-features/models/stored-expressions/). 

**Usage**
In a basic implementation ViewColumn could define a prop for a sql-query which is then autmatically applied to the initial query and returned in the field connected to ViewColumn. This could also be improved to support any kind of select-syntax already existing in TypeORM. Maybe this could also be used for ViewEntity defining all fields in ViewColumns, ViewEntity only defining a basetable if required.

```
@Entity()
export class Report {
    @Field(type => ID)
    public id: number;

    @ViewColumn({sql: `IF (SELECT count(*) FROM data WHERE data.reportId = :id) > 0 { 'inProgress' } ELSE { 'open'}`)
    public state: string;
}
```

**Comment by Ben-Ho** [comment]
I would also try to provide a merge-request for this feature. I'm working with typeorm only for about a month, so it will take some time to understand how this could be implemented. But first of all I want to know if a feature like this is welcome in your library.

**Comment by CollinCashio** [comment]
@Ben-Ho Please see https://github.com/typeorm/typeorm/issues/9323 - I have a pull request out now and it seems like it fits what you are looking for.

**Comment by CollinCashio** [comment]
@pleerock @Ben-Ho @Ginden 

I believe that this should be closed since this feature was implemented in #9323

**Comment by Ben-Ho** [comment]
Looks good to me.
